catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java

```diff
@@ -215,6 +251,109 @@ public int compare(Map<String, Object> o1, Map<String, Object> o2) {
         return metatypes;
     }
 
+    @Override
+    public List<Serializable> updatePublications(String source, List<String> destinations)
+            throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+        List<Serializable> updatedPublishedLocations = new ArrayList<>();
+
+        if (destinations != null) {
+
+            Filter filter = filterBuilder.attribute(RegistryObjectMetacardType.REGISTRY_ID)
+                    .is()
+                    .equalTo()
+                    .text(source);
+            Query query = new QueryImpl(filter);
+
+            QueryResponse queryResponse = catalogFramework.query(new QueryRequestImpl(query));
+
+            List<Result> metacards = queryResponse.getResults();
+            if (metacards != null && metacards.size() > 0) {
+                Metacard metacard = metacards.get(0)
+                        .getMetacard();
+                if (metacard != null) {
+                    List<Serializable> currentlyPublishedLocations = metacard.getAttribute(
+                            RegistryObjectMetacardType.PUBLISHED_LOCATIONS)
+                            .getValues();
+
+                    List<String> unpublishLocations = new ArrayList<>();
+
+                    unpublishLocations = currentlyPublishedLocations.stream()
+                            .filter(location -> !destinations.contains((String) location))
+                            .map(location -> (String) location)
+                            .collect(Collectors.toList());
+
```

Comment:
Let me verify I understand what you are suggesting here: Instead of saying `for (String id : destinations)` I should explicitly avoid things that are already published? That makes sense to me, not sure why I didn't do that in the first place. :)